### PR TITLE
CRS optional with default to OGC:CRS84

### PR DESCRIPTION
**Pull request by alasarr**
This PR aims to find a solution to the discussion of CRS  https://github.com/opengeospatial/geoparquet/issues/52.

It converts the CRS into optional and set the default value to OGC:CRS84.

### CODE REVIEW

**Review by cayetanobv — CHANGES_REQUESTED** [review]

File: format-specs/geoparquet.md [hunk 1]
```diff
@@ -52,21 +52,21 @@ Each geometry column in the dataset must be included in the columns field above
...
+| crs       | string   | **Optional** [WKT2](https://docs.opengeospatial.org/is/18-010r7/18-010r7.html) string representing the Coordinate Reference System (CRS) of the geometry. The default value is [OGC:84](http://www.opengis.net/def/crs/OGC/1.3/CRS84)(long/lat coordinates) |
 | edges | string | **OPTIONAL** Name of the coordinate system for the edges. Must be one of 'planar' or 'spherical'. The default value is 'planar'.  |
 | bbox   | \[number] | **OPTIONAL** Bounding Box of the geometries in the file, formatted according to [RFC 7946, section 5](https://tools.ietf.org/html/rfc7946#section-5) |
 
 #### crs
 
-The Coordinate Reference System (CRS) is a mandatory parameter for each geometry column defined in geoparquet format. 
+The Coordinate Reference System (CRS) is an optional parameter for each geometry column defined in geoparquet format. 
 
 The CRS must be provided in [WKT](https://en.wikipedia.org/wiki/Well-known_text_representation_of_coordinate_reference_systems) version 2, also known as **WKT2**. WKT2 has several revisions, this specification only supports [WKT2_2019](https://docs.opengeospatial.org/is/18-010r7/18-010r7.html).
```
[the remark this answers is not in the document]
Comment (alasarr) [reply format-specs/geoparquet.md:66]: Done

**Review by cayetanobv — APPROVED** [review]

File: examples/example.py [hunk 2]
```diff
@@ -8,24 +8,23 @@
...
    {'columns': {'geometry': {'bbox': [-180.0, -90.0, 180.0, 83.6451],
-                             'crs': 'GEOGCRS["WGS 84",ENSEMBLE["World Geodetic '
-                                    'System 1984 ensemble",MEMBER["World Geodetic '
-                                    'System 1984 (Transit)"],MEMBER["World '
-                                    'Geodetic System 1984 (G730)"],MEMBER["World '
-                                    'Geodetic System 1984 (G873)"],MEMBER["World '
-                                    'Geodetic System 1984 (G1150)"],MEMBER["World '
-                                    'Geodetic System 1984 (G1674)"],MEMBER["World '
-                                    'Geodetic System 1984 (G1762)"],MEMBER["World '
+                             'crs': 'GEOGCRS["WGS 84 (CRS84)",ENSEMBLE["World '
```
Comment (kylebarron) [inline examples/example.py:11]: I find that this output is hard to read because lines are very long... inside of a dict... inside of Python output ... inside of a comment... I also personally don't find it very clean to update this code comment in each PR, because the git diffs are hard to parse.

I'd propose that for our examples we have a Parquet file and then a neighboring JSON file containing the metadata. So for example:

```
example1.parquet
// plain text schema that matches what's stored in example1.parquet
example1_schema.json

example2.parquet
// plain text schema that matches what's stored in example2.parquet
example2_schema.json
```

We could have a CI step that verifies the plain text schema in `*_schema.json` matches what's actually stored in `example1.parquet`.

I think this setup would make it easier for people browsing the spec to see examples of what the schema looks like without needing to open up a file.

Comment (cholmes) [reply examples/example.py:11]: +1 - I like the idea of a neighboring example file that show's what's stored in it.

And yeah, seems in line with the general idea to make JSON schema a part of the core repo. I haven't made an issue for that one either. Feel free to make that issue, and include yours or make a linked issue.

Comment (alasarr) [reply examples/example.py:11]: good idea @kylebarron!

Comment (kylebarron) [reply examples/example.py:11]: I made #65 to discuss this.

File: format-specs/geoparquet.md [hunk 3]
```diff
@@ -76,21 +77,27 @@ GEOGCRS["WGS 84",
...
+        SCOPE["Not known."],
         AREA["World."],
         BBOX[-90,-180,90,180]],
-    ID["EPSG",4326]]
+    ID["OGC","CRS84"]]
 ```
 
-Due to the large number of CRSes available and the difficulty of implementing all of them, we expect that a number of implementations will at least start with only supporting a single CRS. To maximize interoperability we strongly recommend GeoParquet tool providers to always implement support for [EPSG:4326](https://epsg.org/crs_4326/WGS-84.html). 
-Users are recommended to store their data in EPSG:4326 for it to work with the widest number of tools. But data that is better served in particular projections can choose to use an alternate coordinate reference system. We expect many tools will support alternate CRSes, but encourage users to check.
+Due to the large number of CRSes available and the difficulty of implementing all of them, we expect that a number of implementations will at least start with only supporting a single CRS. To maximize interoperability we strongly recommend GeoParquet tool providers to always implement support for [OGC:CRS84](http://www.opengis.net/def/crs/OGC/1.3/CRS84).
```
Comment (Jesus89) [inline format-specs/geoparquet.md:99]: There is a working `https` link. Let's use that instead.

https://www.opengis.net/def/crs/OGC/1.3/CRS84

Offtopic: I noticed that the XML definition files point to the `http` links (for example https://www.opengis.net/def/crs/OGC). Does anyone know why and if it's possible to update those XML files?

```xml
<identifiers xmlns="http://www.opengis.net/crs-nts/1.0" xmlns:gco="http://www.isotc211.org/2005/gco" xmlns:gmd="http://www.isotc211.org/2005/gmd" at="http://www.opengis.net/def/crs/OGC">
<identifier>http://www.opengis.net/def/crs/OGC/0</identifier>
<identifier>http://www.opengis.net/def/crs/OGC/1.3</identifier>
</identifiers>
```

File: format-specs/geoparquet.md [hunk 4]
```diff
@@ -52,22 +52,23 @@ Each geometry column in the dataset must be included in the columns field above
...
 
 #### crs
 
-The Coordinate Reference System (CRS) is a mandatory parameter for each geometry column defined in geoparquet format. 
+The Coordinate Reference System (CRS) is an optional parameter for each geometry column defined in geoparquet format. 
 
 The CRS must be provided in [WKT](https://en.wikipedia.org/wiki/Well-known_text_representation_of_coordinate_reference_systems) version 2, also known as **WKT2**. WKT2 has several revisions, this specification only supports [WKT2_2019](https://docs.opengeospatial.org/is/18-010r7/18-010r7.html).
 
-For the widest interoperability we recommend [EPSG:4326](https://epsg.org/crs_4326/WGS-84.html) for all data, as it is the most widely used coordinate reference system today, so unless data is stored in an alternate projection the CRS should be:
+If CRS is not provided, the default value is [OGC:CRS84](http://www.opengis.net/def/crs/OGC/1.3/CRS84). It's equivalent to the wellknown [EPSG:4326](https://epsg.org/crs_4326/WGS-84.html) but changing the axis from Lat-Long to Long-Lat. WKT2:2019 string for OGC:CRS84 is:
```
Comment (Jesus89) [inline format-specs/geoparquet.md:68]: Same https://github.com/opengeospatial/geoparquet/pull/60/files#r849714833

File: format-specs/geoparquet.md [hunk 5]
```diff
@@ -52,22 +52,23 @@ Each geometry column in the dataset must be included in the columns field above
 
 | Field Name |                               Type                                      |                                                                   Description                                                                     |
 | ---------- | ----------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------- |
-| crs       | string   | **REQUIRED** [WKT2](https://docs.opengeospatial.org/is/18-010r7/18-010r7.html) string representing the Coordinate Reference System (CRS) of the geometry.  |
-| epoch    | double | **OPTIONAL** Coordinate epoch in case of a dynamic CRS, expressed as a decimal year.  |
 | encoding | string | **REQUIRED** Name of the geometry encoding format. Currently only 'WKB' is supported. |
+| crs       | string   | **OPTIONAL** [WKT2](https://docs.opengeospatial.org/is/18-010r7/18-010r7.html) string representing the Coordinate Reference System (CRS) of the geometry. The default value is [OGC:CRS84](http://www.opengis.net/def/crs/OGC/1.3/CRS84)(long/lat coordinates) |
```
Comment (Jesus89) [inline format-specs/geoparquet.md:56]: Same https://github.com/opengeospatial/geoparquet/pull/60/files#r849714833

File: format-specs/geoparquet.md [hunk 6]
```diff
@@ -76,21 +77,27 @@ GEOGCRS["WGS 84",
...
-Users are recommended to store their data in EPSG:4326 for it to work with the widest number of tools. But data that is better served in particular projections can choose to use an alternate coordinate reference system. We expect many tools will support alternate CRSes, but encourage users to check.
+Due to the large number of CRSes available and the difficulty of implementing all of them, we expect that a number of implementations will at least start with only supporting a single CRS. To maximize interoperability we strongly recommend GeoParquet tool providers to always implement support for [OGC:CRS84](http://www.opengis.net/def/crs/OGC/1.3/CRS84). 
+Users are recommended to store their data in OGC:CRS84 (long/lat) for it to work with the widest number of tools. But data that is better served in particular projections can choose to use an alternate coordinate reference system. We expect many tools will support alternate CRSes, but encourage users to check.
```
Comment (Jesus89) [inline format-specs/geoparquet.md:100]: I would use `OGC:CRS84 (longitude-latitude)` to make the spec more explicit

File: format-specs/geoparquet.md (hunk 5 above)
Comment (Jesus89) [reply format-specs/geoparquet.md:56]: Same https://github.com/opengeospatial/geoparquet/pull/60/files#r849721020

File: format-specs/geoparquet.md (hunk 4 above)
Comment (Jesus89) [reply format-specs/geoparquet.md:68]: Same https://github.com/opengeospatial/geoparquet/pull/60/files#r849721020

File: format-specs/geoparquet.md (hunk 3 above)
Comment (cholmes) [reply format-specs/geoparquet.md:99]: You mean that they are using http:// instead of https:// ? 

@ogcscotts should be able to help, or at least find the person to help.

**Review by Jesus89 — APPROVED** [review]

File: format-specs/geoparquet.md (hunk 3 above)
Comment (ogcscotts) [reply format-specs/geoparquet.md]: I am checking with the latest discussions and conversations in the OGC Naming Authority... more very soon.

Comment (ogcscotts) [reply format-specs/geoparquet.md]: update: for OGC Registries, we have revised policy for https (https://github.com/opengeospatial/NamingAuthority/issues/108#issuecomment-1047735391)

for schemas.opengis.net, the Architecture Domain Working Group will be meeting in the next weeks and into June to work out a policy and implementation for http and https.

Comment (Jesus89) [reply format-specs/geoparquet.md]: Sounds good. Thanks!

File: format-specs/geoparquet.md [hunk 7]
```diff
@@ -52,22 +52,23 @@ Each geometry column in the dataset must be included in the columns field above
 
 | Field Name |                               Type                                      |                                                                   Description                                                                     |
 | ---------- | ----------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------- |
-| crs       | string   | **REQUIRED** [WKT2](https://docs.opengeospatial.org/is/18-010r7/18-010r7.html) string representing the Coordinate Reference System (CRS) of the geometry.  |
-| epoch    | double | **OPTIONAL** Coordinate epoch in case of a dynamic CRS, expressed as a decimal year.  |
 | encoding | string | **REQUIRED** Name of the geometry encoding format. Currently only 'WKB' is supported. |
+| crs       | string   | **OPTIONAL** [WKT2](https://docs.opengeospatial.org/is/18-010r7/18-010r7.html) string representing the Coordinate Reference System (CRS) of the geometry. The default value is [OGC:CRS84](https://www.opengis.net/def/crs/OGC/1.3/CRS84) (longitude-latitude coordinates) |
```
Comment (cholmes) [inline format-specs/geoparquet.md:56]:
```suggestion
| crs       | string   | **OPTIONAL** [WKT2](https://docs.opengeospatial.org/is/18-010r7/18-010r7.html) string representing the Coordinate Reference System (CRS) of the geometry. If the crs field is not included then the data in this column must be stored in longitude, latitude. In the case where a crs is not provided, CRS-aware implementations should assume a default value of [OGC:CRS84](https://www.opengis.net/def/crs/OGC/1.3/CRS84) (longitude-latitude coordinates) |
```

File: format-specs/geoparquet.md [hunk 8]
```diff
@@ -52,22 +52,23 @@ Each geometry column in the dataset must be included in the columns field above
...
 
 #### crs
 
-The Coordinate Reference System (CRS) is a mandatory parameter for each geometry column defined in geoparquet format. 
+The Coordinate Reference System (CRS) is an optional parameter for each geometry column defined in geoparquet format. 
 
 The CRS must be provided in [WKT](https://en.wikipedia.org/wiki/Well-known_text_representation_of_coordinate_reference_systems) version 2, also known as **WKT2**. WKT2 has several revisions, this specification only supports [WKT2_2019](https://docs.opengeospatial.org/is/18-010r7/18-010r7.html).
 
-For the widest interoperability we recommend [EPSG:4326](https://epsg.org/crs_4326/WGS-84.html) for all data, as it is the most widely used coordinate reference system today, so unless data is stored in an alternate projection the CRS should be:
+If CRS is not provided, the default value is [OGC:CRS84](https://www.opengis.net/def/crs/OGC/1.3/CRS84). It's equivalent to the wellknown [EPSG:4326](https://epsg.org/crs_4326/WGS-84.html) but changing the axis from latitude-longitude to longitude-latitude. WKT2:2019 string for OGC:CRS84 is:
```
Comment (cholmes) [inline format-specs/geoparquet.md:68]:
```suggestion
If CRS is not provided, then all coordinates in the geometry must use longitude, latitude to store their data.
If an implementation is CRS-aware and needs a CRS representation of the data it should assume a default value is [OGC:CRS84](https://www.opengis.net/def/crs/OGC/1.3/CRS84). It's equivalent to the well-known [EPSG:4326](https://epsg.org/crs_4326/WGS-84.html) but changes the axis from latitude-longitude to longitude-latitude. The WKT2:2019 string for OGC:CRS84 is:
```

File: format-specs/geoparquet.md [hunk 9]
```diff
@@ -76,21 +77,27 @@ GEOGCRS["WGS 84",
...
+        SCOPE["Not known."],
         AREA["World."],
         BBOX[-90,-180,90,180]],
-    ID["EPSG",4326]]
+    ID["OGC","CRS84"]]
 ```
 
-Due to the large number of CRSes available and the difficulty of implementing all of them, we expect that a number of implementations will at least start with only supporting a single CRS. To maximize interoperability we strongly recommend GeoParquet tool providers to always implement support for [EPSG:4326](https://epsg.org/crs_4326/WGS-84.html). 
-Users are recommended to store their data in EPSG:4326 for it to work with the widest number of tools. But data that is better served in particular projections can choose to use an alternate coordinate reference system. We expect many tools will support alternate CRSes, but encourage users to check.
+Due to the large number of CRSes available and the difficulty of implementing all of them, we expect that a number of implementations will at least start with only supporting a single CRS. To maximize interoperability we strongly recommend GeoParquet tool providers to always implement support for [OGC:CRS84](https://www.opengis.net/def/crs/OGC/1.3/CRS84).
```
Comment (cholmes) [inline format-specs/geoparquet.md:99]:
```suggestion
Due to the large number of CRSes available and the difficulty of implementing all of them, we expect that a number of implementations will start without support for the optional `crs` field. 
```

File: format-specs/geoparquet.md [hunk 10]
```diff
@@ -76,21 +77,27 @@ GEOGCRS["WGS 84",
...
-Users are recommended to store their data in EPSG:4326 for it to work with the widest number of tools. But data that is better served in particular projections can choose to use an alternate coordinate reference system. We expect many tools will support alternate CRSes, but encourage users to check.
+Due to the large number of CRSes available and the difficulty of implementing all of them, we expect that a number of implementations will at least start with only supporting a single CRS. To maximize interoperability we strongly recommend GeoParquet tool providers to always implement support for [OGC:CRS84](https://www.opengis.net/def/crs/OGC/1.3/CRS84). 
+Users are recommended to store their data in OGC:CRS84 (longitude-latitude) for it to work with the widest number of tools. But data that is better served in particular projections can choose to use an alternate coordinate reference system. We expect many tools will support alternate CRSes, but encourage users to check.
```
Comment (cholmes) [inline format-specs/geoparquet.md:100]:
```suggestion
Users are recommended to store their data in longitude, latitude (OGC:CRS84 or not including the `crs` field) for it to work with the widest number of tools. But data that is better served in particular projections can choose to use an alternate coordinate reference system. We expect many tools will support alternate CRSes, but encourage users to check to ensure their chosen tool supports their chosen crs.
```

**Review by cholmes — COMMENTED** [review]
Have a few tweaks on language, but otherwise looks good. Leaving this as a comment so others can merge, will mark 'approve' once these commit.

**Comment by alasarr** [comment]
@cholmes thanks a lot for the suggestions! I've just committed them

**Comment by cayetanobv** [comment]
> Have a few tweaks on language, but otherwise looks good. Leaving this as a comment so others can merge, will mark 'approve' once these commit.

@cholmes I agree with your comments. Thanks!

**Review by cholmes — APPROVED** [review]
